From 48050dbcb7d74f3f3b7294ae2fb35d70e94c47d2 Mon Sep 17 00:00:00 2001
From: Alessandro Fontani <alessandrofontani@gmail.com>
Date: Wed, 30 May 2018 16:52:03 +0200
Subject: [PATCH] disabilitazione autonegoziazione u-boot


diff --git a/common/main.c b/common/main.c
index c27f123..b53b6aa 100644
--- a/common/main.c
+++ b/common/main.c
@@ -41,7 +41,7 @@ static void run_preboot_environment_command(void)
 }
 
 
-#define SHOW_ENGICAM_VERSION     "Version: Engicam Yocto U-Boot 3.00\n"
+#define SHOW_ENGICAM_VERSION     "Version: Contact Yocto U-Boot 3.00 Autoneg Off\n"
 
 //! This function show on u-boot consolle on start-up the version
 //  For setting the version open file mx25_3stack.
diff --git a/drivers/net/fec_mxc.c b/drivers/net/fec_mxc.c
index 1c558e5..8e867b4 100644
--- a/drivers/net/fec_mxc.c
+++ b/drivers/net/fec_mxc.c
@@ -983,6 +983,7 @@ static int fec_probe(bd_t *bd, int dev_id, uint32_t base_addr,
 	unsigned char ethaddr[6];
 	uint32_t start;
 	int ret = 0;
+	unsigned short ReadReg;
 
 	/* create and fill edev struct */
 	edev = (struct eth_device *)malloc(sizeof(struct eth_device));
@@ -1044,6 +1045,35 @@ static int fec_probe(bd_t *bd, int dev_id, uint32_t base_addr,
 #endif
 	eth_register(edev);
 
+        /* DND Modifica Autonegoziazione OFF */
+	
+	printf("\nBOOT AUTOMDIX DISABLE\n");
+	
+	//printf("\nPhyId = %d\n",phydev->phy_id);	
+	ReadReg = fec_mdio_read(bus->priv, 0, 4);
+	//printf("MDIO READ 04 BEFORE = %02x\n",ReadReg);
+	ReadReg |= (1 << 10);
+	fec_mdio_write(bus->priv, 0, 4, ReadReg);
+	ReadReg = fec_mdio_read(bus->priv, 0, 4);
+	//printf("MDIO READ 04 AFTER = %02x\n",ReadReg);
+
+	ReadReg = fec_mdio_read(bus->priv, 0, 27);
+	//printf("MDIO READ 27 BEFORE = %02x\n",ReadReg);
+	ReadReg |= (1 << 15);
+	fec_mdio_write(bus->priv, 0, 27, ReadReg);
+	ReadReg = fec_mdio_read(bus->priv, 0, 27);
+	//printf("MDIO READ 27 AFTER = %02x\n",ReadReg);
+
+	printf("BOOT AUTONEG DISABLE\n");
+
+	ReadReg = fec_mdio_read(bus->priv, 0, 0);
+	//printf("MDIO READ 0 BEFORE = %02x\n",ReadReg);
+	ReadReg = 0x2100;
+	fec_mdio_write(bus->priv, 0, 0, ReadReg);
+	ReadReg = fec_mdio_read(bus->priv, 0, 0);
+	//printf("MDIO READ 0 AFTER = %02x\n",ReadReg);
+
+
 	if (fec_get_hwaddr(edev, dev_id, ethaddr) == 0) {
 		debug("got MAC%d address from fuse: %pM\n", dev_id, ethaddr);
 		memcpy(edev->enetaddr, ethaddr, 6);
-- 
2.7.4

